src/common/reactNativeProjectHelper.ts

```diff
@@ -10,53 +10,112 @@ import {Package} from "./node/package";
 import {ErrorHelper} from "../common/error/errorHelper";
 import {InternalErrorCode} from "../common/error/internalErrorCode";
 
+export interface ParsedPackageName {
+    packageName: string;
+    useSemverCoerce: boolean;
+}
+
+export interface RNPackageVersions {
+    reactNativeVersion: string;
+    reactNativeWindowsVersion: string;
+}
+
 export class ReactNativeProjectHelper {
 
     public static getRNVersionsWithBrokenMetroBundler() {
         // https://github.com/Microsoft/vscode-react-native/issues/660 for details
         return ["0.54.0", "0.54.1", "0.54.2", "0.54.3", "0.54.4"];
     }
 
-    public static getReactNativeVersion(projectRoot: string): Q.Promise<string> {
-        return ReactNativeProjectHelper.getReactNativePackageVersionFromNodeModules(projectRoot)
+    public static getReactNativeVersions(projectRoot: string, isRNWindows: boolean = false): Q.Promise<RNPackageVersions> {
+        return ReactNativeProjectHelper.getReactNativePackageVersionsFromNodeModules(projectRoot, isRNWindows)
             .catch(err => {
-                return ReactNativeProjectHelper.getReactNativeVersionFromProjectPackage(projectRoot);
+                return ReactNativeProjectHelper.getReactNativeVersionsFromProjectPackage(projectRoot, isRNWindows);
             });
     }
 
-    public static getReactNativePackageVersionFromNodeModules(projectRoot: string): Q.Promise<string> {
-        return new Package(projectRoot).dependencyPackage("react-native").version()
-            .then(version => ReactNativeProjectHelper.processVersion(version))
-            .catch(err => {
-                throw ErrorHelper.getInternalError(InternalErrorCode.ReactNativePackageIsNotInstalled);
-            });
+    public static getReactNativePackageVersionsFromNodeModules(projectRoot: string, isRNWindows: boolean = false): Q.Promise<RNPackageVersions> {
+        let versionPromises: Q.Promise<string>[] = [];
+
+        versionPromises.push(
+            new Package(projectRoot).getPackageVersionFromNodeModules("react-native")
```

Comment:
Can we move that part of the code into a private function, so we could prepare such a promise without code duplicating?